src/app/modules/webhook/webhook.service.ts

```diff
@@ -217,3 +224,51 @@ const getFormattedResponse = (
     },
   }
 }
+
+export const pushData = (
+  webhookUrl: WebhookParams['webhookUrl'],
+  submissionWebhookView: WebhookView,
+): Promise<any> => {
+  const now = Date.now()
+  // Log and return, this should not happen.
+  if (!submissionWebhookView) {
+    logWebhookFailure(
+      new WebhookValidationError('submissionWebhookView was null'),
+      {
+        webhookUrl,
+        submissionWebhookView,
+        now,
+      },
+    )
+    return
+  }
+
+  const { submissionId, formId } = submissionWebhookView.data
+
+  const signature = formsgSdk.webhooks.generateSignature({
+    uri: webhookUrl,
+    submissionId,
+    formId,
+    epoch: now,
+  }) as string
+
+  const webhookParams = {
+    webhookUrl,
+    submissionWebhookView,
+    submissionId,
+    formId,
+    now,
+    signature,
+  }
+
+  // Use promises instead of await to prevent the user from having to await on the
+  // webhook before they receive acknowledgement that their submission was successful.
+  return validateWebhookUrl(webhookParams.webhookUrl)
```

Comment:
We can still use async/await nomenclature here, they are fundamentally the same thing. Just need to make sure the caller of this function does not `await` this function

Prefer async/await unless absolutely necessary

Reply:
done